File: data/gpt-4o/readable_format_HPEC/cpp/file277.cpp
#include <stdio.h>
#include <omp.h>

int a[100];
void gen_task(int i);

int main()
{
    int i;

#pragma omp parallel
 {
#pragma omp single
  {
   for(i=0; i<100; i++)
    gen_task(i);
  }
 }

 for(i=0; i<100; i++)
  if(a[i]!=(i+1))
   printf("warning: a[%d] = %d not expected %d
",i,a[i],i+1);
 return 0;
}

void gen_task(int i)
{
#pragma omp task
 a[i]=i+1;
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file277.f90
program DRB101_task_value_orig_no
    use omp_lib
    use DRB101
    implicit none

    integer :: i
    allocate (a(100))

    !$omp parallel
        !$omp single
        do i = 1, 100
            call gen_task(i)
        end do
        !$omp end single
    !$omp end parallel

    do i = 1, 100
        if (a(i) /= i+1) then
            print*,'warning: a(',i,') =',a(i),' not expected',i+1
        end if
!        print*,a(i),i+1
    end do
end program
